### ISSUE TITLE
Multiple server calls when sorting is enabled

### ISSUE THREAD

**Issue by dor-squadded** [grid-js/gridjs#1356]
**Describe the bug**
Multiple server calls when sorting is enabled.

**To Reproduce**
1. Go to https://gridjs.io/docs/examples/server-side-sort
2. Open dev tools on the network tab.
3. Use the sort on any column.
4. See that the calls to the server are doubled when sorting.
5. Comment out the `sort: false` on the images column.
6. Use the sort on any column.
8. See that the calls to the server are tripled.

**Expected behavior**
The server calls should happen only once, independant of the number of column you can sort.

**Screenshots**
1. 
![image](https://github.com/grid-js/gridjs/assets/76150116/2777721a-0c0b-4734-b352-4c401ee7a060)
2. 
![image](https://github.com/grid-js/gridjs/assets/76150116/69c8c3bf-2193-4f4a-bf55-17739296b01b)

**Desktop (please complete the following information):**
 - OS: Windows 11.
 - Browser: Chrome
 - Version: 113

**Additional context**
This can be problematic when dealing with a table that has 5+ columns that need to be sorted. The requests will be 5X more than needed!

**Comment by explody** [comment]
I just noticed something similar myself, though I'm getting an even weirder behavior:

* sort, pagination, search are false: 2 identical queries to the server URL
* sort, pagination are false, search is server: 3 identical queries to the server
* if search,sort is true and pagination false, I get 5 queries.
* if all three are true, I get *six* queries. The first to the baser server URL, then the next 5 including 'limit' and 'page' but all identical

In a nutshell, setting these to server for me results in anywhere from 2-6 effectively identical queries, depending on the combo.

**Comment by cyrilccs** [comment]
I'm facing the same issue, and it's quite concerning that such an important problem is still persisting despite the number of users of the solution. 
The fact that the server is receiving 5/10 identical queries is highly inefficient.

I kindly request community help in resolving this matter.

@afshinm

**Comment by dor-kuperman** [comment]
@cyrilccs Agreed, it's an issue that needs to be fixed ASAP

**Comment by dsentker** [comment]
I can confirm this issue too with version > 6.x
Using 5.1.0 solves the issue.

**Comment by cornonthecobb** [comment]
I am also experiencing the same issue when I do the server side post call instead of the get that uses a request body instead of the url.

**Comment by aswzen** [comment]
Any update on this?

**Comment by happybananaDev** [comment]
I have the same problem,6 columns = 6 server calls

**Comment by vinthefantastic1** [comment]
I thought I was doing something wrong. I see the same issue using 6.0.6

**Comment by vkresch** [comment]
Same issue. Any updates?

**Comment by jakecausier** [comment]
Issue also happens to me. With 10 columns, it's hammering my server with 10 requests at once!

**Comment by PLJean** [comment]
Any update on this one? This is a serious issue my server.

**Comment by PLJean** [comment]
Any update on this one? I can confirm it's happening to me as well.

**Comment by BastienTwist** [comment]
Same for me with v6.0.6

**Comment by afshinm** [comment]
Apologies for this issue. I have a fix for this that should address this issue and improve the overall rendering performance of the library: https://github.com/grid-js/gridjs/pull/1408

**Comment by dsentker** [comment]
you don't need to apologize, it's an open source library where any experienced developer could have contributed. Thank you for taking care of it 🙏

**Comment by afshinm** [comment]
A new version is now available. Please try v6.1.0 https://www.npmjs.com/package/gridjs
